src/Test/QuickCheck/StateMachineModel.hs

```diff
@@ -36,26 +40,33 @@ import           Text.PrettyPrint.ANSI.Leijen            (Pretty, align, dot,
 
 import           Test.QuickCheck.StateMachineModel.Utils
 
+import           Unsafe.Coerce
+
 ------------------------------------------------------------------------
 
 liftSem
   :: forall cmd resp ix ref m
   .  (Monad m, Enum ix, Typeable ref, RefKit cmd)
   => Typeable resp
+  => Ord (ix, Int)
   => Show resp
   => (cmd resp ref -> m resp)
-  -> cmd resp ix -> StateT [ref] m resp
-liftSem sem cmd = do
+  -> Int
```

Comment:
Yep, I wanted to do this next. :-)

Probably something like CommandWithPid cmd, so we can avoid the Enum instance for pairs...